src/core/server/status/routes/status.ts

```diff
@@ -0,0 +1,198 @@
+    };
+    response_times: {
+      max_in_millis: number;
+    };
+    requests: {
+      total: number;
+      disconnects: number;
+      statusCodes: Record<number, number>;
+      status_codes: Record<number, number>;
+    };
+    concurrent_connections: number;
+  };
+}
+
+export const registerStatusRoute = ({ router, config, metrics, status }: Deps) => {
+  // Since this observable is not subscribed to elsewhere, we need to subscribe
+  // here to eagerly load the plugins status when Kibana starts up.
+  const plugins$ = new ReplaySubject<Record<PluginName, ServiceStatus>>();
+  status.plugins$.subscribe(plugins$);
+
+  router.get(
+    {
+      path: '/api/status',
+      options: {
+        authRequired: !config.allowAnonymous,
+        tags: ['api'], // ensures that unauthenticated calls receive a 401 rather than a 302 redirect to login page
+      },
+      validate: {
+        query: schema.object({
+          v8format: schema.boolean({ defaultValue: false }),
+        }),
+      },
+    },
+    async (context, req, res) => {
+      const { version, buildSha, buildNum } = config.packageInfo;
+      const versionWithoutSnapshot = version.replace(SNAPSHOT_POSTFIX, '');
+      const [overall, core, plugins] = await combineLatest([
+        status.overall$,
+        status.core$,
+        plugins$,
+      ])
+        .pipe(first())
+        .toPromise();
+
+      let statusInfo: StatusInfo | LegacyStatusInfo;
+      if (req.query?.v8format) {
+        statusInfo = {
+          overall: serializeStatus(overall),
+          core: serializeStatusRecord((core as unknown) as Record<string, ServiceStatus>),
+          plugins: serializeStatusRecord(plugins),
+        };
+      } else {
+        statusInfo = calculateLegacyStatus({
+          overall,
+          core,
+          plugins,
+          versionWithoutSnapshot,
+        });
+      }
+
+      const lastMetrics = await metrics.getOpsMetrics$().pipe(first()).toPromise();
+
+      const body: StatusHttpBody = {
+        name: config.serverName,
+        uuid: config.uuid,
+        version: {
+          number: versionWithoutSnapshot,
+          build_hash: buildSha,
+          build_number: buildNum,
+          build_snapshot: SNAPSHOT_POSTFIX.test(version),
+        },
+        status: statusInfo,
+        metrics: {
+          last_updated: lastMetrics.collected_at.toISOString(),
+          collection_interval_in_millis: metrics.collectionInterval,
+          os: lastMetrics.os,
+          process: lastMetrics.process,
+          response_times: lastMetrics.response_times,
+          concurrent_connections: lastMetrics.concurrent_connections,
+          requests: {
+            ...lastMetrics.requests,
+            status_codes: lastMetrics.requests.statusCodes,
+          },
+        },
+      };
+
+      return res.ok({ body });
+    }
+  );
+};
+
+const serializeStatus = (status: ServiceStatus): SerializableServiceStatus => ({
+  ...status,
+  level: status.level.toString(),
```

Comment:
`status.level` can implement [toJSON](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/JSON/stringify#toJSON_behavior) in order to able to use `JSON.stringify` instead of implementing a manual serialiser.